CONTRIBUTING.md

```diff
@@ -345,6 +345,18 @@ with clear instructions on how to install the required libraries.
 Furthermore, if your package requires CGO, it MUST be able to compile and operate in a no op mode
 or report a warning back to the collector with a clear error saying CGO is required to work.
 
+### Breaking changes
```

Comment:
Question: What is the expected way to change a func signature (breaking change, but assuming it just returns 2 values instead of 1) in the new "world":
1. Create a new `func` and deprecate the old one - downside probably we will not use the best name for the func since that is the old func. Can also be solved by doing this in multiple steps, add a temporary (not ideal name), remove the deprecated one, deprecate the temporary and reuse the good name, remove temporary.
2. Accept some "breaking changes" without deprecation.

deleted the initial comment, and moved it here to allow threads :D

Reply:
How would we do it after v1? We'd likely have a new function name that is composed by the ideal name + differences in behavior. Like:

- step 1 (current API): `func GetFoo() Foo`
- step 2 (current+1): needs to return error, then:
- `func MustGetFoo() Foo` (panics on error, is the replacement for `GetFoo() Foo`, which gets deprecated in favor of the new MustGetFoo
- step 3 (current+2): `func GetFoo() (Foo, error)` is added, `func MustGetFoo() Foo` is not touched

Reply:
If we have agreement, I'll add this example to the doc.

Reply:
Fine with this. After v1 I will have to survive with the bad name :))